diff --git a/js/posting/posting.js b/js/posting/posting.js
index 313b09d..3a8c1f2 100755
--- a/js/posting/posting.js
+++ b/js/posting/posting.js
@@ -41,8 +41,9 @@ function spotPosting() {
 				console.log('error: '+((new XMLSerializer()).serializeToString(xml)));
 			}
 		});
+	} // cbHashcashCalculated
 		
-		this.rpHashcashCalculated = function (self, hash) {
+	this.rpHashcashCalculated = function (self, hash) {
 			self.reportForm['postreportform[newmessageid]'].value = hash;
 			self.reportForm['postreportform[submit]'].value = 'Post';
 			self.uiDone();
@@ -55,21 +56,22 @@ function spotPosting() {
 				dataType: "xml",
 				data: dataString2,  
 				success: function(xml) {
+			alert(((new XMLSerializer()).serializeToString(xml)));
 					var result = $(xml).find('result').text();
 					if(result == 'success') {
 						var user = $(xml).find('user').text();
 						var userid = $(xml).find('userid').text();
 						var text = $(xml).find('body').text();
 						var useridurl = 'http://'+window.location.hostname+window.location.pathname+'?search[tree]=&amp;search[type]=UserID&amp;search[text]='+userid;
-					}
+					} else {
+					console.log('error: '+((new XMLSerializer()).serializeToString(xml)));
+					} // else					
 				},
 				error: function(xml) {
 					console.log('error: '+((new XMLSerializer()).serializeToString(xml)));
 				}
 			});
-		
-		
-	} // callback
+	} // callback rpHashcashCalculated
 
 	this.postComment = function(commentForm, uiStart, uiDone) {
 		this.commentForm = commentForm;
@@ -78,7 +80,7 @@ function spotPosting() {
 		
 		// update the UI 
 		this.uiStart();
-		
+
 		// First retrieve some values from the form we are submitting
 		var randomstr = commentForm['postcommentform[randomstr]'].value;
 		var rating = commentForm['postcommentform[rating]'].value;
@@ -90,10 +92,10 @@ function spotPosting() {
 
 		/* Nu vragen we om, asynchroon, een hashcash te berekenen. Zie comments van calculateCommentHashCash()
 		   waarom dit asynhcroon verloopt */
-		this.calculateCommentHashCash('<' + inreplyto + '.' + rating + '.' + randomstr, '@spot.net>', 0, this.cbHashcashCalculated);
+		this.calculateCommentHashCash('<' + inreplyto + '.' + rating + '.' + randomstr + '.', '@spot.net>', 0, this.cbHashcashCalculated);
 	} // postComment
 	
-	this.postReport = function(reportForm) {
+	this.postReport = function(reportForm, uiStart, uiDone) {
 		this.reportForm = reportForm;
 		this.uiStart = uiStart;
 		this.uiDone = uiDone;
@@ -105,7 +107,7 @@ function spotPosting() {
 		var inreplyto = reportForm['postreportform[inreplyto]'].value;
 		inreplyto = inreplyto.substring(0, inreplyto.indexOf('@'));
 		
-		this.calculateCommentHashCash('<' + inreplyto + '.' + randomstr, '@spot.net>', 0, this.rpHashcashCalculated);
+		this.calculateCommentHashCash('<' + inreplyto + '.' + randomstr + '.', '@spot.net>', 0, this.rpHashcashCalculated);
 	} // postReport
 	
 	//
@@ -126,12 +128,12 @@ function spotPosting() {
 				uniquePart += possibleChars.charAt(irand);
 			} // for
 
-			hash = $.sha1(prefix + '.' + uniquePart + suffix);
+			hash = $.sha1(prefix + uniquePart + suffix);
 			validHash = (hash.substr(0, 4) == '0000');
 		} while ((!validHash) && ((runCount % 500) != 0));
 
 		if (validHash) {
-			cbWhenFound(this, prefix + '.' + uniquePart + suffix);
+			cbWhenFound(this, prefix + uniquePart + suffix);
 		} else {
 			if (runCount > 400000) {
 				alert("Unable to calculate SHA1 hash: " + runCount);
diff --git a/lib/SpotDb.php b/lib/SpotDb.php
index 085deee..0de20aa 100755
--- a/lib/SpotDb.php
+++ b/lib/SpotDb.php
@@ -95,6 +95,18 @@ class SpotDb {
 		return (empty($tmpResult));
 	} # isCommentMessageIdUnique
 
+	/* 
+	 * Controleer of een messageid niet al eerder gebruikt is door ons om hier
+	 * te posten
+	 */
+	function isReportMessageIdUnique($messageid) {
+		$tmpResult = $this->_conn->singleQuery("SELECT messageid FROM reportsposted WHERE messageid = '%s'",
+						Array($messageid));
+		
+		return (empty($tmpResult));
+	} # isReportMessageIdUnique
+
+
 	/*
 	 * Sla het gepostte comment op van deze user
 	 */
@@ -116,7 +128,7 @@ class SpotDb {
 	 */
 	function addPostedReport($userId, $report) {
 		$this->_conn->modify(
-				"INSERT INTO reportssposted(ouruserid, messageid, inreplyto, randompart, body, stamp)
+				"INSERT INTO reportsposted(ouruserid, messageid, inreplyto, randompart, body, stamp)
 					VALUES('%d', '%s', '%s', '%s', '%s', %d)", 
 				Array((int) $userId,
 					  $report['newmessageid'],
@@ -340,30 +352,6 @@ class SpotDb {
 								WHERE id = '%s'", 
 							Array( (int) $userid));
 	} # deleteUser
-	
-	/*
-	 * Verwijder spots en comments van een user uit de db
-	 */
-	function removeUser($userId) {
-		switch ($this->_dbsettings['engine']) {
-			case 'pdo_pgsql'  : 
-			case 'pdo_sqlite' : {
-				$this->_conn->modify("DELETE FROM spots WHERE messageid IN (SELECT messageid FROM spotsfull WHERE userid = '%s')", Array($userId));
-				$this->_conn->modify("DELETE FROM commentsxover WHERE nntpref IN (SELECT messageid FROM spotsfull WHERE userid= '%s')", Array($userId));
-				$this->_conn->modify("DELETE FROM spotsfull WHERE userid = '%s')", Array($userId));
-				$this->_conn->modify("DELETE FROM commentsfull WHERE userid = '%s')", Array($userId));
-				break;
-			} # pdo_sqlite
-			
-			default			: {
-				$this->_conn->modify("DELETE FROM spotsfull, commentsfull, spots, commentsxover USING spotsfull
-										LEFT JOIN commentsfull ON spotsfull.userid=commentsfull.userid
-										LEFT JOIN spots ON spotsfull.messageid=spots.messageid
-										LEFT JOIN commentsxover ON spotsfull.messageid=commentsxover.messageid
-										WHERE spotsfull.userid = '%s'", Array($userId));
-			} # default
-		} # switch	
-	} # removeUser
 
 	/*
 	 * Update de informatie over een user behalve het password
diff --git a/lib/SpotNntp.php b/lib/SpotNntp.php
index f4ef79e..be1c0e3 100755
--- a/lib/SpotNntp.php
+++ b/lib/SpotNntp.php
@@ -548,7 +548,7 @@ class SpotNntp {
 			return $spot;
 		} # getFullSpot 
 		
-		function reportSpotAsSpam($user, $serverPrivKey, $title, $report) {
+		function reportSpotAsSpam($user, $serverPrivKey, $newsgroup, $title, $report) {
 			# instantieer de benodigde objecten
 			$spotSigning = new SpotSigning();
 			$spotParser = new SpotParser();
@@ -561,7 +561,7 @@ class SpotNntp {
 			
 			$header = 'From: ' . $user['username'] . " <" . trim($user['username']) . '@spot.net>' . "\r\n";
 			$header .= 'Subject: REPORT ' . $report['inreplyto'] . ' ' . $title . "\r\n";
-			$header .= 'Newsgroups: free.willey' . "\r\n";
+			$header .= 'Newsgroups: ' . $newsgroup . "\r\n";
 			$header .= 'Message-ID: <' . $report['newmessageid'] . '@spot.net>' . "\r\n";
 			$header .= 'References: <' . $report['inreplyto'] . ">\r\n";
 			$header .= 'X-User-Signature: ' . $spotParser->specialString($user_signature['signature']) . "\r\n";
diff --git a/lib/SpotPosting.php b/lib/SpotPosting.php
index 9c5562f..b1571fe 100755
--- a/lib/SpotPosting.php
+++ b/lib/SpotPosting.php
@@ -118,6 +118,7 @@ class SpotPosting {
 		if (empty($errorList)) {
 			$this->_nntp_post->reportSpotAsSpam($user,
 										   $this->_settings->get('privatekey'),  # Server private key
+										   $this->_settings->get('report_group'),
 										   $fullSpot['title'], 
 										   $report);
 			$this->_db->addPostedReport($user['userid'], $dbReport);
diff --git a/lib/SpotSecurity.php b/lib/SpotSecurity.php
index 3ce6388..6c3822b 100755
--- a/lib/SpotSecurity.php
+++ b/lib/SpotSecurity.php
@@ -58,8 +58,6 @@ class SpotSecurity {
 	const spotsec_keep_own_filters				= 34;
 	const spotsec_set_filters_as_default		= 35;
 	const spotsec_report_spam					= 36;
-	const spotsec_use_modpanel					= 37;
-	const spotsec_rem_spot						= 38;
 	
 	// Array mapping the security id to a human readable text
 	private $_secHumanReadable = array(
@@ -98,10 +96,8 @@ class SpotSecurity {
 		32 		=> "Notificaties versturen (per type)",
 		33		=> "Eigen CSS stylesheet toestaan per gebruiker",
 		34		=> "Zelf spotfilters kunnen bewerken",
-		35		=> "Filters als standaard in kunnen stellen voor nieuwe gebruikers"
-		36		=> "Rapporteer spot of gebruiker als spam",
-		37		=> "Toegang tot moderatorpaneel",
-		38		=> "Spots verwijderen"
+		35		=> "Filters als standaard in kunnen stellen voor nieuwe gebruikers",
+		36		=> "Rapporteer spot of gebruiker als spam"
 	);
 	
 	function __construct(SpotDb $db, SpotSettings $settings, array $user) {
diff --git a/lib/SpotSettingsUpgrader.php b/lib/SpotSettingsUpgrader.php
index d60b92e..8cef433 100755
--- a/lib/SpotSettingsUpgrader.php
+++ b/lib/SpotSettingsUpgrader.php
@@ -110,7 +110,6 @@ class SpotSettingsUpgrader {
 		$this->setIfNot('nzb_group', 'alt.binaries.ftd');
 		$this->setIfNot('comment_group', 'free.usenet');
 		$this->setIfNot('report_group', 'free.willey');
-		$this->setIfNot('report_group', 'free.willey');
 	} # setupNewsgroups()
 	
 } # SpotSettingsUpgrader
diff --git a/lib/SpotUserUpgrader.php b/lib/SpotUserUpgrader.php
index 1a08d2d..ecdce08 100755
--- a/lib/SpotUserUpgrader.php
+++ b/lib/SpotUserUpgrader.php
@@ -108,7 +108,6 @@ class SpotUserUpgrader {
 		$dbCon->rawExec("INSERT INTO usergroups(userid,groupid,prio) VALUES(2, 1, 1)");
 		$dbCon->rawExec("INSERT INTO usergroups(userid,groupid,prio) VALUES(2, 2, 2)");
 		$dbCon->rawExec("INSERT INTO usergroups(userid,groupid,prio) VALUES(2, 3, 3)");
-		$dbCon->rawExec("INSERT INTO usergroups(userid,groupid,prio) VALUES(2, 4, 4)");
 	} # createAdmin
 
 	/*
@@ -319,10 +318,7 @@ class SpotUserUpgrader {
 		
 		# Spam reporting en moderator panel toegevoegd
 		if ($this->_settings->get('securityversion') < 0.15) {
-			$dbCon->rawExec("INSERT INTO securitygroups(id,name) VALUES(4, 'Moderators')");
 			$dbCon->rawExec("INSERT INTO grouppermissions(groupid,permissionid) VALUES(2, " . SpotSecurity::spotsec_report_spam . ")");
-			$dbCon->rawExec("INSERT INTO grouppermissions(groupid,permissionid) VALUES(3, " . SpotSecurity::spotsec_use_modpanel . ")");
-			$dbCon->rawExec("INSERT INTO grouppermissions(groupid,permissionid) VALUES(4, " . SpotSecurity::spotsec_use_modpanel . ")");
 		} # if
 	} # updateSecurityGroups
 
diff --git a/lib/dbstruct/SpotStruct_abs.php b/lib/dbstruct/SpotStruct_abs.php
index b0e8654..377a491 100755
--- a/lib/dbstruct/SpotStruct_abs.php
+++ b/lib/dbstruct/SpotStruct_abs.php
@@ -246,8 +246,6 @@ abstract class SpotStruct_abs {
 		$this->dropForeignKey('commentsposted', 'messageid', 'spots', 'messageid', 'ON DELETE CASCADE ON UPDATE CASCADE');
 		$this->dropForeignKey('commentsxover', 'messageid', 'spots', 'messageid', 'ON DELETE CASCADE ON UPDATE CASCADE');
 		$this->dropForeignKey('commentsfull', 'messageid', 'spots', 'messageid', 'ON DELETE CASCADE ON UPDATE CASCADE');
-		$this->dropForeignKey('reportsposted', 'inreplyto', 'spots', 'messageid', 'ON DELETE CASCADE ON UPDATE CASCADE');
-		$this->dropForeignKey('reportsposted', 'messageid', 'spots', 'messageid', 'ON DELETE CASCADE ON UPDATE CASCADE');
 		
 		##############################################################################################
 		# Opschonen data #############################################################################
diff --git a/lib/page/SpotPage_reportpost.php b/lib/page/SpotPage_reportpost.php
old mode 100644
new mode 100755
index 9329d9b..826e0e0
--- a/lib/page/SpotPage_reportpost.php
+++ b/lib/page/SpotPage_reportpost.php
@@ -54,6 +54,10 @@ class SpotPage_reportpost extends SpotPage_Abs {
 			# zorg er voor dat alle variables ingevuld zijn
 			$report = array_merge($report, $this->_reportForm);
 
+			# het messageid krijgen we met <>'s, maar we werken 
+			# in spotweb altijd zonder, dus die strippen we
+			$report['newmessageid'] = substr($report['newmessageid'], 1, -1);
+			
 			# vraag de users' privatekey op
 			$this->_currentSession['user']['privatekey'] = 
 				$this->_db->getUserPrivateRsaKey($this->_currentSession['user']['userid']);
@@ -64,10 +68,8 @@ class SpotPage_reportpost extends SpotPage_Abs {
 			
 			if (empty($formMessages['errors'])) {
 				$postResult = array('result' => 'success',
-									'user' => $this->_currentSession['user']['username'],
-									'userid' => $spotSigning->calculateUserid($this->_currentSession['user']['publickey']),
-									'body' => $report['body']
-									);
+									'user' => $this->_currentSession['user']['username']
+								   );
 			} else {
 				$postResult = array('result' => 'failure');
 			} # else
diff --git a/lib/retriever/SpotRetriever_Reports.php b/lib/retriever/SpotRetriever_Reports.php
old mode 100644
new mode 100755
diff --git a/templates/we1rdo/adminpanel.inc.php b/templates/we1rdo/adminpanel.inc.php
index 3172ad1..6d9db50 100755
--- a/templates/we1rdo/adminpanel.inc.php
+++ b/templates/we1rdo/adminpanel.inc.php
@@ -11,9 +11,6 @@
 			<li><a href="?page=render&tplname=listusers" title="Gebruikerslijst"><span>Gebruikerslijst</span></a></li>
 <?php } ?>
 			<li><a href="?page=render&tplname=listgroups" title="Groepenlijst"><span>Groepenlijst</span></a></li>
-<?php if ($tplHelper->allowed(SpotSecurity::spotsec_use_modpanel, '')) { ?>
-			<li><a href="?page=render&tplname=modpage" title="Moderatorsectie"><span>Moderatorsectie</span></a></li>
-<?php } ?>	
 		</ul>
 
 		<div id="adminpaneltab-1" class="ui-tabs-hide">
diff --git a/templates/we1rdo/js/scripts.js b/templates/we1rdo/js/scripts.js
index 50d3eb6..a245545 100755
--- a/templates/we1rdo/js/scripts.js
+++ b/templates/we1rdo/js/scripts.js
@@ -28,6 +28,8 @@ $(function(){
 			return true;
 		}
     });	
+	
+	postReportForm();
 });
 
 // createBaseURL
@@ -317,7 +319,7 @@ function loadComments(messageid,perpage,pagenr) {
 
 function postReportForm() {
 	$("form.postreportform").submit(function(){ 
-		new spotPosting().postReport(this); 
+		new spotPosting().postReport(this,postCommentUiStart,postCommentUiDone); 
 		return false;
 	});	
 }
diff --git a/templates/we1rdo/mod/last5.php b/templates/we1rdo/mod/last5.php
deleted file mode 100644
index 32d8701..0000000
--- a/templates/we1rdo/mod/last5.php
+++ /dev/null
@@ -1,20 +0,0 @@
-<?php
-$n = 1;
-$i = 1234;
-
-while($n <= 5)
-{
-echo "<tr>";
-echo "<form action=\"lrac.php\" method=\"post\">";
-echo "<td name=\"title\">";
-echo "<input type=\"hidden\" name=\"title\" value=\"Test #" . $n . ", wordt een script\">Test #" . $n . ", wordt een script</td>";
-echo "<td name=\"spotid\">" . $i . "</td>";
-echo "<td name=\"repid\">" . $n . "</td>";
-echo "<td name=\"spotloc\">" . $i . "@spotnet.nl</td>";
-echo "<td><select name=\"opt\"><option value=\"rems\">Remove Spot</option><option value=\"remr\">Ignore Report</option></select></td><td><input type=\"submit\" value=\"Ok\"></td>";
-echo "</form>";
-echo "</tr>";
-$n++;
-$i += 1111;
-}
-?>
diff --git a/templates/we1rdo/mod/lrac.php b/templates/we1rdo/mod/lrac.php
deleted file mode 100644
index 2fcaf0e..0000000
--- a/templates/we1rdo/mod/lrac.php
+++ /dev/null
@@ -1,5 +0,0 @@
-<html>
-<body>
-Je hebt gekozen voor <?php echo $_POST["opt"]; ?> in <?php echo $_POST["title"]; ?>!
-</body>
-</html>
\ No newline at end of file
diff --git a/templates/we1rdo/mod/style.css b/templates/we1rdo/mod/style.css
deleted file mode 100644
index 25ebe11..0000000
--- a/templates/we1rdo/mod/style.css
+++ /dev/null
@@ -1,16 +0,0 @@
-.quickfunc
-{
-position:absolute;
-right:20px;
-top:20px;
-}
-
-.quickbutton
-{
-width:150px;
-}
-
-.lastrep
-{
-top:80px;
-}
diff --git a/templates/we1rdo/modpage.inc.php b/templates/we1rdo/modpage.inc.php
deleted file mode 100644
index abfffe1..0000000
--- a/templates/we1rdo/modpage.inc.php
+++ /dev/null
@@ -1,23 +0,0 @@
-<html>
-<head>
-<link rel="stylesheet" type="text/css" href="mod/style.css" />
-</head>
-<body>
-<h3>Welcome at The Moderator Page</h3>
-<table class="quickfunc">
-<tr><th colspan="3"><h3>Quick Functions</h3></th></tr>
-<tr><td><input type="button" class="quickbutton" value="Get External Reports"></td><td><input type="button" class="quickbutton" value="Ban User"></td></tr>
-<tr><td><input type="button" class="quickbutton" value="View External Reports"></td><td><input type="button" class="quickbutton" value="View Local Reports"></td></tr>
-<tr><td><a href="templates/we1rdo/mod/rem1.html"><input type="button" class="quickbutton" value="Remove Spot"></td><td><a href="templates/we1rdo/mod/rem2.html"><input type="button" class="quickbutton" value="Remove User"></td><td></td></tr>
-</table>
-
-<table class="lastrep" cellspacing="5">
-<tr><th span="5">Last five reports</th></tr>
-<tr><th>Title</th><th>Spot ID</th><th>Report ID</th><th>NNTP Address</th></tr>
-<?php
-include_once "mod/last5.php";
-
-?>
-</table>
-</body>
-</html>
diff --git a/templates/we1rdo/remspot.inc.php b/templates/we1rdo/remspot.inc.php
deleted file mode 100644
index f976f82..0000000
--- a/templates/we1rdo/remspot.inc.php
+++ /dev/null
@@ -1,19 +0,0 @@
-<?php
-if ($_SERVER['REQUEST_METHOD']=="GET" && $_GET['id']){
-require_once "lib/SpotClassAutoload.php"; 
-require_once "settings.php"; 
-require_once "lib/SpotTiming.php"; 
-require_once "lib/exceptions/ParseSpotXmlException.php"; 
-require_once "lib/exceptions/NntpException.php";
-
-try {
-	$db = new SpotDb($settings['db']); 
-	$db->connect(); 
-} catch(Exception $x) {
-	die("Unable to connect to database: " . $x->getMessage() . PHP_EOL); 
-} # catch
-
-deleteSpot($_GET['id']);
-
-} #if
-?>
\ No newline at end of file
diff --git a/templates/we1rdo/spamreport.inc.php b/templates/we1rdo/spamreport.inc.php
old mode 100644
new mode 100755
diff --git a/templates/we1rdo/spotinfo.inc.php b/templates/we1rdo/spotinfo.inc.php
index e1d476e..3087b07 100755
--- a/templates/we1rdo/spotinfo.inc.php
+++ b/templates/we1rdo/spotinfo.inc.php
@@ -8,7 +8,6 @@
 						 ($tplHelper->allowed(SpotSecurity::spotsec_retrieve_nzb, ''))
 						);
 	$show_watchlist_button = ($currentSession['user']['prefs']['keep_watchlist'] && $tplHelper->allowed(SpotSecurity::spotsec_keep_own_watchlist, ''));
-	$show_rem_spot = ($tplHelper->allowed(SpotSecurity::spotsec_rem_spot, ''));
 ?>
 		<div id="details" class="details <?php echo $tplHelper->cat2color($spot) ?>">
 			<table class="spotheader">
@@ -17,15 +16,6 @@
 						<th class="back"> <a class="closeDetails" title="Ga terug naar het overzicht (esc / u)">&lt;&lt;</a> </th>
 						<th class="category"><span><?php echo $spot['formatname'];?></span></th>
 						<th class="title"><?php echo $spot['title'];?></th>
-						<th class="remSpot">
-							<?php 
-								if ($tplHelper->allowed(SpotSecurity::spotsec_rem_spot, '')) {
-								include "remspot.inc.php";
-								$msgId = $spot['messageid'];
-								?>
-								<a class="remSpot" onClick="remSpot('<?php echo htmlspecialchars($msgId); ?>')"> </a>
-							<?php } #if ?>
-						</th>
 						<th class="rating">
 <?php
 	if($spot['rating'] == 0) {
